memory/mm_subsystem.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2020 IBM
+# Author: Harish <harish@linux.ibm.com>
+
+import os
+from avocado import Test
+from avocado.utils import build
+from avocado.utils import memory
+from avocado.utils import process
+from avocado.utils import git
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class MmSubsystemTest(Test):
+    '''
+    Memory subsystem functional test
+
+    :avocado: tags=memory
+    '''
+    @staticmethod
+    def clear_dmesg():
+        process.run("dmesg -C", sudo=True)
+
+    @staticmethod
+    def check_dmesg():
+        errorlog = ['WARNING: CPU:', 'Oops', 'Segfault', 'soft lockup',
+                    'ard LOCKUP', 'Unable to handle paging request',
+                    'rcu_sched detected stalls', 'NMI backtrace for cpu']
+        err = []
+        logs = process.system_output("dmesg -Txl 1,2,3,4"
+                                     "").decode("utf-8").splitlines()
+        for error in errorlog:
+            for log in logs:
+                if error in log:
+                    err.append(log)
+        return "\n".join(err)
+
+    def setUp(self):
+        """
+        Build binary and filter tests based on the environment
+        """
+        smm = SoftwareManager()
+        deps = ['gcc', 'make', 'patch']
+        detected_distro = distro.detect()
+        if detected_distro.name in ["Ubuntu", 'debian']:
+            deps.extend(['libpthread-stubs0-dev', 'git'])
+        elif detected_distro.name == "SuSE":
+            deps.extend(['glibc-devel-static', 'git-core'])
+        else:
+            deps.extend(['glibc-static', 'git', 'runc'])
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+        git.get_repo("https://gitlab.com/cailca/linux-mm",
+                     destination_dir=self.logdir)
+        os.chdir(self.logdir)
+        if detected_distro.name == "SuSE":
+            suse_patch = self.fetch_asset("1.patch", locations=[
+                "https://patch-diff.githubusercontent.com/raw/harish-24/"
```

Comment:
is this location available as  i am not able to locate via browser and wget also not works 
```
[praveen@MiWiFi-R3L-srv ~]$ wget https://patch-diff.githubusercontent.com/raw/harish-24/linux-mm/pull/.patch
--2020-08-27 13:55:15--  https://patch-diff.githubusercontent.com/raw/harish-24/linux-mm/pull/.patch
Resolving patch-diff.githubusercontent.com (patch-diff.githubusercontent.com)... 13.234.176.102
Connecting to patch-diff.githubusercontent.com (patch-diff.githubusercontent.com)|13.234.176.102|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
Cookie coming from patch-diff.githubusercontent.com attempted to set domain to github.com
2020-08-27 13:55:15 ERROR 404: Not Found.

[praveen@MiWiFi-R3L-srv ~]$ 
```

Reply:
Thanks for catching this. It seems to be a typo (missing "1") when editing indentation I guess, fixing it now.

Reply:
ah ok sure